### ISSUE TITLE
Duplicate or repetitive notifications during recalculation

### ISSUE THREAD

**Issue by zyphlar** [organicmaps/organicmaps#6927]
**Describe the issue**
During rerouting sometimes a turn instruction can get repeated even after the system should know that the instruction is no longer valid. Previously this would manifest as extraneous "turn left. Turn left. Turn right. Turn left" type instructions when you've missed a planned then, but now with street name announcement in the beta it's more obvious that some of these are just "old" and inaccurate.

**Steps to reproduce**
Navigate in a dense urban area with lots of alternate routes, like a US city's "grid." Plan a route into the middle of that grid so there's at least a few turns with a few options in a short distance.

Upon reaching, for example, "turn left on Main Street," continue straight instead. Maybe turn left or right one block after Main Street. Alternately, turn left or right one block *before* Main Street, preempting the notification by a few hundred feet.

Notice that the TTS output is something like "turn left on Main Street. Rerouting. Turn left on Main Street. Turn right on 2nd Avenue."

In other words, the system should already have discarded the Main Street turn but it appears to have been cached or brought along as part of the new route erroneously. A correct clean reroute will say "turn left on Main Street. Rerouting. Turn right on 2nd Avenue" to catch up with the new reality of your drive.
